indigo_app/views/works.py

```diff
@@ -1346,16 +1345,69 @@ def get_context_data(self, **kwargs):
             repeal_made = self.request.POST.get('repeal_made')
             if repeal_made:
                 repealed_work = Work.objects.filter(pk=repeal_made).first()
-                if repealed_work:
+                if repealed_work and repealed_work.repealed_by != self.work and not any(
+                        [True for i in initial if i["repealed_work"] == repealed_work]):
                     initial.append({
-                         'repealed_work': repealed_work,
-                         'repealed_date': repealed_work.repealed_date,
+                        'repealed_work': repealed_work,
+                        'repealed_date': repealed_work.repealed_date or self.work.commencement_date,
                     })
 
         context["form"] = {
-           'repeals_made_formset': RepealMadeBaseFormSet(prefix='repeals_made', initial=initial, form_kwargs={"work": self.work}),
+           'repeals_made_formset': RepealMadeBaseFormSet(
+               prefix='repeals_made',
+               initial=initial,
+               form_kwargs={"work": self.work}
+           ),
         }
         context["work"] = self.work
         return context
 
 
+class WorkFormAmendmentsView(WorkViewBase, TemplateView):
+    template_name = 'indigo_api/_work_form_amendments_form.html'
+
+    def post(self, request, *args, **kwargs):
+        return self.get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        formset = AmendmentsBaseFormSet(self.request.POST, prefix="amendments", form_kwargs={"work": self.work})
+        initial = []
+        if formset.is_valid():
+            for form in formset:
+                delete = form.cleaned_data.get('DELETE')
+                if delete:
+                    if not Amendment.objects.filter(
+                            amended_work=self.work,
+                            amending_work=form.cleaned_data["amending_work"]
+                    ).exists():
+                        continue
+                initial.append({
+                    "amended_work": self.work,
+                    "amending_work": form.cleaned_data["amending_work"],
+                    "date": form.cleaned_data["date"],
+                    "DELETE": form.cleaned_data["DELETE"],
+                })
+            amendment = self.request.POST.get("amendment")
+            if amendment:
+                amending_work = Work.objects.filter(pk=amendment).first()
+                if amending_work:
+                    # add amendment only if it does not exist and is not in the initial data
+                    if not Amendment.objects.filter(amended_work=self.work, amending_work=amending_work).exists() and not any(
+                            [True for i in initial if i["amending_work"] == amending_work]):
```

Comment:
hmmmmmm does this imply that we can't add more than one amendment by the same work? or am I misreading.

Reply:
yeah, thats what I was doing. I take it (from this comment and the ones below) that we are allowed to have more than one amendment by the same work? just not at the same date?